gammapy/data/ivoa.py

```diff
@@ -0,0 +1,338 @@
+
+        if obs_publisher_did is None:
+
+            log.warning(
+                "Insufficient publisher information: 'obs_publisher_did' obscore value will be empty."
+            )
+        if access_url is None:
+            log.warning(
+                "Insufficient publisher information: access_url' obscore value will be empty."
+            )
+    else:
+        obs_publisher_did = ""
+        access_url = ""
+        log.warning(
+            "Insufficient publisher information: 'obs_publisher_did' and 'access_url' obscore values will be empty."
+        )
+
+    tab = _obscore_def()
+    tab.add_row()
+
+    observation = data_store.obs(single_obsID)
+
+    obs_mask = data_store.obs_table["OBS_ID"] == observation.obs_id
+    obs_pos = data_store.obs_table[obs_mask]["EVENTS_FILENAME"]
+    path = make_path(os.environ["GAMMAPY_DATA"] + "/" + obs_pos[0])
+    size = int(os.path.getsize(path) / 1000.0)
+    tab["dataproduct_type"] = observation.obs_info["EXTNAME"]
+    tab["calib_level"] = 2  # Look into the data
+    tab["target_name"] = observation.obs_info["OBJECT"]
+    tab["obs_id"] = str(observation.obs_info["OBS_ID"])
+    tab["obs_collection"] = "DL3"
+    tab["obs_publisher_did"] = obs_publisher_did
+    tab["access_url"] = access_url
+    tab["access_format"] = "application/fits"
+    tab["access_estsize"] = size
+    tab["s_ra"] = observation.get_pointing_icrs(observation.tmid).ra.value
+    tab["s_dec"] = observation.get_pointing_icrs(observation.tmid).dec.value
+    tab["s_fov"] = 10.0
+    tab["t_min"] = observation.tstart.value
+    tab["t_max"] = observation.tstop.value
+    tab["t_exptime"] = observation.observation_live_time_duration.value
+    tab["em_min"] = observation.events.energy.min().value
+    tab["em_max"] = observation.events.energy.max().value
+    tab["facility_name"] = observation.obs_info["TELESCOP"]
+    tab["instrument_name"] = observation.obs_info["TELLIST"]
+
+    return tab
+
+
+def to_obscore_table(base_dir, selected_obs=None, **kwargs):
+
+    """Generate the complete obscore Table by stacking length=1 Tables given by DataStore._obscore_row()
+
+    Parameters
+    ----------
+    selected_obs : list or array of Observation ID (int)
+    (default of ``None`` means ``no obaservation ``)
+    If not given, the obscore default table is returned.
+
+
+
+    **kwargs : `str` {obs_publisher_did, access_url}
+    Giving the values for is highly recommended.
+    If any of these are not given the corresponding obscore field is left empty and a warning is raised for each empty value.
+
+    Returns
+    -------
+    Astropy Table length = len(selected_obs)
+    """
+
+    tab_default = _obscore_def()
+
+    if selected_obs is None:
+        obscore_tab = tab_default
+
+    else:
+        obscore_tab = Table()
+        for i in range(0, len(selected_obs)):
+            obscore_row = _obscore_row(base_dir, selected_obs[i], **kwargs)
+            obscore_tab = vstack([obscore_tab, obscore_row])
```

Comment:
I find the logic a bit puzzling. Why do you create a table and fill it with one element at each step e.g. with `Table.add_row()`

Reply:
Thanks for your reviews. In fact, using a table "vstack" was the only way I found in the beginning to be able to achieve the IVOA requests where they recommend keeping the empty "values" consistent with each column data type. Now, because of how the script evolved, it seems reasonable to me to do it using .add_row(). 

I pushed a new version of ivoa.py where now _obscore_row() does not generate a table of length “one” but directly adds a row to the already existing table for each ObsID.